src/main/java/sqlitedb/SQLiteDB.java

```diff
@@ -3,25 +3,40 @@
 import org.apache.commons.logging.Log;
 import org.apache.commons.logging.LogFactory;
 import org.sqlite.SQLiteConnection;
+import sandbox.ArchivableFile;
 import sandbox.SqlSandboxUtils;
 import services.ConnectionHandler;
 
 import javax.sql.DataSource;
-import java.io.File;
+import java.io.IOException;
 import java.sql.Connection;
 import java.sql.SQLException;
 
 public class SQLiteDB implements ConnectionHandler {
     private DBPath dbPath;
+    private ArchivableFile dbArchivableFile;
     private final Log log = LogFactory.getLog(SQLiteDB.class);
     private Connection connection;
 
     public SQLiteDB(DBPath dbPath) {
         this.dbPath = dbPath;
+        /*
+           FormplayerStorageFactory and RestoreFactory are instantiated with sqLiteDB = SQLLiteDB(null)
+           and sqLiteDB is only set to a real value during .configure;
+           this doesn't make a lot of sense to me, but appears to be required for tests.
+        */
+        if (dbPath != null) {
+            dbArchivableFile = new ArchivableFile(dbPath.getDatabaseFile());
+        }
     }
 
     private Connection getNewConnection() throws SQLException {
-        DataSource dataSource = SqlSandboxUtils.getDataSource(dbPath.getDatabaseName(), dbPath.getDatabasePath());
+        try {
+            dbArchivableFile.unarchiveIfArchived();
+        } catch (IOException e) {
+            throw new RuntimeException(e);
```

Comment:
Yeah, I'm not totally sure. There are a number of method calls within `unarchiveIfArchived` that have the potential to throw an `IOException`, and I'd rather see the error happen and then deal with each kind of situation specifically then silence all potential errors. For example, let's say we silenced the error, and the specific potential error you mentioned does happen,maybe even systematically. Well in that case, we'll be constantly deleting and rebuilding these database, but there will be no indication of that, only a grinding slowness in an already slow system that will be hard to locate or pinpoint.

On the other hand, if we do get an error, we can go straight to the place, see the weird state, and try and figure out exactly why it happened, how widespread it is, etc. and then decide how we want to handle it.

I guess you could make the case to report it as an error, but do automatic cleanup. I'm actually unfamiliar with the tools we'd use in java/formplayer to do that, but I'm also just tempted to leave it be and deal with it if and when it becomes an issue